File: benchmarks/tsb/bench_dataframe_astype.ts
import { DataFrame } from "tsb";

const rng = (seed: number) => { let s = seed; return () => { s = (s * 1664525 + 1013904223) & 0xffffffff; return (s >>> 0) / 0xffffffff; }; };
const rand = rng(42);
const df = new DataFrame({
  a: Array.from({ length: 100_000 }, () => (rand() * 2 - 1) * 3),
  b: Array.from({ length: 100_000 }, () => Math.floor(rand() * 1000)),
});
for (let i = 0; i < 3; i++) df.astype({ a: "float32", b: "int32" });
const N = 100;
const t0 = performance.now();
for (let i = 0; i < N; i++) df.astype({ a: "float32", b: "int32" });
const elapsed = performance.now() - t0;
console.log(JSON.stringify({ function: "dataframe_astype", mean_ms: elapsed / N, iterations: N, total_ms: elapsed }));

File: benchmarks/pandas/bench_dataframe_astype.py
import pandas as pd, json, time, numpy as np
rng = np.random.default_rng(42)
df = pd.DataFrame({"a": rng.standard_normal(100_000), "b": rng.integers(0, 1000, size=100_000)})
for _ in range(3): df.astype({"a": "float32", "b": "int32"})
N = 100
t0 = time.perf_counter()
for _ in range(N): df.astype({"a": "float32", "b": "int32"})
elapsed = time.perf_counter() - t0
print(json.dumps({"function": "dataframe_astype", "mean_ms": elapsed/N*1000, "iterations": N, "total_ms": elapsed*1000}))
